sdk/src/main/java/com/horizen/node/NodeState.java

```diff
@@ -1,8 +1,14 @@
 package com.horizen.node;
 
+import com.horizen.params.NetworkParams;
 import com.horizen.state.SidechainStateReader;
+import com.horizen.utils.WithdrawalEpochInfo;
 
 public interface NodeState extends NodeStateBase, SidechainStateReader {
 
     boolean hasCeased();
+
+    NetworkParams params();
+
+    WithdrawalEpochInfo getWithdrawalEpochInfo();
```

Comment:
why do we need this new interfaces?
I tempted to remove them from `NodeState`

Reply:
SidechainNodeView uses a NodeState type in it that we cannot override, and we cannot use other View type cause  SidechainApiRoute has a SidechainNodeViewBase type in it, which is only extended by SidechainNodeView.

Its another mess with types, I can show you in a bit more detail on a call